optuna/study/_tell.py

```diff
@@ -180,7 +179,7 @@ def _tell_with_warning(
     finally:
         study._storage.set_trial_state_values(frozen_trial._trial_id, state, values)
 
-    frozen_trial = copy.deepcopy(study._storage.get_trial(frozen_trial._trial_id))
+    frozen_trial = study._storage.get_trial(frozen_trial._trial_id)
 
     if warning_message is not None:
         frozen_trial._system_attrs[STUDY_TELL_WARNING_KEY] = warning_message
```

Comment:
`frozen_trial` is modified in this line. So, could you refactor the warning message implementation first?